File: 3-LongestSubstringWithoutRepeatingCharacters/code.py
class Solution:
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        maxx = tmp = 0
        queue = []
        for ss in s:
            if ss not in queue:
                tmp += 1
            else:
                if tmp > maxx:
                    maxx = tmp
                while queue[0] != ss:          
                    del queue[0]
                    tmp -=1
                del queue[0]
            queue.append(ss)    
        if tmp > maxx:
            return tmp
        return maxx
                

File: 637-Average_of_Levels_in_Binary_Tree/code.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        queue<TreeNode* > queue;
        vector<double> res;
        queue.push(root);
        while(! queue.empty()){
            long tmp = 0;
            int size = queue.size();
            for (int i = 0; i < size; ++i){
                TreeNode* t = queue.front();
                queue.pop();
                if (t -> left){
                    queue.push(t -> left);
                }
                if (t -> right){
                    queue.push(t -> right);
                }
                tmp += t -> val;
            }
            res.push_back((double)tmp/size);
        }
        return res;
    }
};